src/util/artifacts.js

```diff
@@ -14,12 +16,41 @@ export async function prepareArtifactsDir(
     const stats = await fs.stat(artifactsDir);
     if (!stats.isDirectory()) {
       throw new UsageError(
-        `--artifacts-dir=${artifactsDir} (this value is not a directory)`);
+        `--artifacts-dir="${artifactsDir}" exists but it is not a directory.`);
+    }
+    // If the artifactsDir already exists, check that we have the write permissions on it.
+    try {
+      await fs.access(artifactsDir, fs.W_OK);
+    } catch (accessErr) {
+      if (isErrorWithCode('EACCES', accessErr)) {
+        throw new UsageError(
+          `--artifacts-dir="${artifactsDir}" exists but the user lacks ` +
+          'permissions on it.');
+      } else {
+        throw accessErr;
+      }
     }
   } catch (error) {
-    if (isErrorWithCode('ENOENT', error)) {
-      log.debug(`Creating artifacts directory: ${artifactsDir}`);
-      await fs.mkdir(artifactsDir);
+    if (isErrorWithCode('EACCES', error)) {
+      // Handle errors when the artifactsDir cannot be accessed.
+      throw new UsageError(
+        `Cannot access --artifacts-dir="${artifactsDir}" because the user ` +
+        'lacks permissions.');
+    } else if (isErrorWithCode('ENOENT', error)) {
+      // Create the artifact dir if it doesn't exist yet.
+      try {
+        log.debug(`Creating artifacts directory: ${artifactsDir}`);
+        await asyncMkdirp(artifactsDir);
+      } catch (mkdirErr) {
+        if (isErrorWithCode('EACCES', mkdirErr)) {
+          // Handle errors when the artifactsDir cannot be created for lack of permissions.
+          throw new UsageError(
+            `Cannot create --artifacts-dir="${artifactsDir}" because the ` +
+            'user lacks permissions.');
+        } else {
+          throw mkdirErr;
```

Comment:
@saintsebastian being able to reproduce in a test case the `ENOSPC` scenario is not as "easy" as it was with the "lack of permissions" scenarios, and so it is reasonable to use the the usual "dependency injection strategy", this time restricted to this single parameter and this single test.